apps/api/src/domain/models/term/tokenization/ChilcotinAlphabetParser.ts

```diff
@@ -0,0 +1,259 @@
+// import { isNonEmptyObject } from '@coscrad/validation-constraints';
+// import { InternalError } from 'apps/api/src/lib/errors/InternalError';
+
+import assert = require('node:assert');
+import { InternalError } from '../../../../lib/errors/InternalError';
+import { AlphabetCharacters } from './tokenizer.interface';
+
+class Node {
+    text: string;
+
+    transitions: Map<string, Node> = new Map();
+
+    isLetter: boolean;
```

Comment:
I suppose this is essentially a prefix tree, but using the Chilcotin letters (possibly multiple Latin symbols like "tl") instead.